src/Controller/Component/PaginatorComponent.php

```diff
@@ -188,6 +190,9 @@ public function paginate($object, array $settings = [])
         $results = $query->all();
         $numResults = count($results);
         $count = $numResults ? $query->count() : 0;
+        if ($options['maxPage'] && $count > $options['maxPage'] * $options['limit']) {
```

Comment:
The maxLimit is actually the one restricting the "total limit" of records. limit is only for display per page.